### PR TITLE
Feat(contracts): Add script + guide to set aggregator address

### PR DESCRIPTION
**Pull request by PatStiles**
# Add script + guide to set aggregator address
## Description

 Adds script and guide to set the aggregator address of the AlignedServiceManager.sol contract

### To Test
- Start local anvil node.
```bash
make anvil_start_with_block_time
```bash
- Check the current value of `alignedAggregator` within the AlignedServiceManager.sol contract.
```bash
cast call 0x851356ae760d987E095750cCeb3bC6014560891C "alignedAggregator()(address)"
```
- Set the value of `alignedAggregator` using the script.
```bash
export AGGREGATOR_ADDRESS=0x0000000000000000000000000000000000000001
make set_aggregator_address_devnet
```
- Verify `alignedAggregator` has been changed to `0x0000000000000000000000000000000000000001` within the AlignedServiceManager.sol contract.
```bash
cast call 0x851356ae760d987E095750cCeb3bC6014560891C "alignedAggregator()(address)"
```

## Type of change

Please delete options that are not relevant.

- [x] New feature

## Checklist

- [ ] “Hotfix” to `testnet`, everything else to `staging`
- [ ] Linked to Github Issue
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]() 
- [ ] If your PR changes the Operator compatibility (Ex: Upgrade prover versions)
  - [ ] This PR adds compatibility for operator for both versions and do not change batcher/docs/examples
  - [ ] This PR updates batcher and docs/examples to the newer version. This requires the operator are already updated to be compatible

### CODE REVIEW

File: docs/0_internal/setting_aggregator_address.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Setting Aggregator Address
+This doc contains a guide on how to call `setAggregator(address)` to set the `alignedAggregator` value of the AlignedServiceManager.sol contract. 
+
+### NOTE:
+- This guide assumes the Aligned layer contracts have been sucessfully deployed and the deployment outputs are within `./contracts/script/output/<DEPLOYMENT_FOLDER>
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
- This guide assumes the Aligned layer contracts have been sucessfully deployed and the deployment outputs are within `./contracts/script/output/<DEPLOYMENT_FOLDER>`
```

File: docs/0_internal/setting_aggregator_address.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+The address of Aligned Aggregator can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within 
+```
+    "permissions": {
+        ...
+        "alignedLayerAggregator": "<AGGREGATOR_ADDRESS",
+        ...
+    }
+```
+
+## Locate the Aligend Service Manager Address
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## Locate the Aligned Service Manager Address
```

File: docs/0_internal/setting_aggregator_address.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Setting Aggregator Address
+This doc contains a guide on how to call `setAggregator(address)` to set the `alignedAggregator` value of the AlignedServiceManager.sol contract. 
+
+### NOTE:
+- This guide assumes the Aligned layer contracts have been sucessfully deployed and the deployment outputs are within `./contracts/script/output/<DEPLOYMENT_FOLDER>
+
+## Locate the deployed Aligned Aggregator Address
+
+The address of Aligned Aggregator can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
The address of Aligned Aggregator can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within:
```

File: docs/0_internal/setting_aggregator_address.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "permissions": {
+        ...
+        "alignedLayerAggregator": "<AGGREGATOR_ADDRESS",
+        ...
+    }
+```
+
+## Locate the Aligend Service Manager Address
+
+The address of Aligned Service Manager can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
The address of Aligned Service Manager can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within:
```

File: docs/0_internal/setting_aggregator_address.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+export ALIGNED_SERVICE_MANAGER_ADDRESS=<aligned_service_manager_address>
+```
+
+## Check the current value of `alignedAggregator` within AlignedServiceManager.sol
+
+```
+cast call --rpc-url <RPC_URL> $ALIGNED_SERVICE_MANAGER_ADDRESS "alignedAggregator()(address)"
+```
+
+You should see that the printed address matches the returned address matches the address from `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json`
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
You should see that the returned address matches the address from `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` 
```

Comment (avilagaston9) [reply docs/0_internal/setting_aggregator_address.md]: You could also add steps numbers. Then you can easily reference previous steps for example:
```Markdown
You should see that the returned address matches the address from step `1`.
```

File: docs/0_internal/setting_aggregator_address.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+## Change the value of `alignedAggregator` within AlignedServiceManager.sol
+
+Set the environment variable `AGGREGATOR_ADDRESS` to the new address of the aggregator.
+```
+export AGGREGATOR_ADDRESS=<aggregator_address>
+make set_aggregator_address
+```
+
+## Verify the Aligend Aggreagtor Address has changed
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## Verify the Aligned Aggregator Address has changed
```

File: docs/0_internal/setting_aggregator_address.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    }
+```
+
+## Locate the Aligend Service Manager Address
+
+The address of Aligned Service Manager can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within
+```
+    "addresses": {
+        ...
+        "alignedLayerServiceManager": "<ALIGNED_SERVICE_MANAGER_ADDRESS",
```
Comment (JulianVentura) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
        "alignedLayerServiceManager": "<ALIGNED_SERVICE_MANAGER_ADDRESS>",
```

File: Makefile [hunk 8]
```diff
@@ -600,6 +600,16 @@ upgrade_add_aggregator: ## Add Aggregator to Aligned Contracts
 	@echo "Adding Aggregator to Aligned Contracts..."
 	@. contracts/scripts/.env && . contracts/scripts/upgrade_add_aggregator_to_service_manager.sh
 
+set_aggregator_address: ## Add Aggregator to Aligned Contracts
```
Comment (JuArce) [inline Makefile]: This should use NETWORK parameter as used in #1429 (https://github.com/yetanotherco/aligned_layer/pull/1429/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R567)

**Review by JuArce — CHANGES_REQUESTED** [review]

File: docs/0_internal/setting_aggregator_address.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    }
+```
+
+## Set Environment Variables
+
+To run the make targets specified in this guide, you must first set the following env vars within `./contracts/scripts/.env`:
+```
+export RPC_URL=<rpc_url>
+export PRIVATE_KEY=<aligned_service_manager_pauser_private_key>
+export ALIGNED_SERVICE_MANAGER_ADDRESS=<aligned_service_manager_address>
```
Comment (JulianVentura) [inline docs/0_internal/setting_aggregator_address.md]: I think we don't need the exports here:

```suggestion
RPC_URL=<rpc_url>
PRIVATE_KEY=<aligned_service_manager_pauser_private_key>
ALIGNED_SERVICE_MANAGER_ADDRESS=<aligned_service_manager_address>
```

**Review by JulianVentura — COMMENTED** [review]
Worked fine locally, left you some suggestions

File: docs/0_internal/setting_aggregator_address.md (hunk 6 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 5 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed! @JulianVentura

File: docs/0_internal/setting_aggregator_address.md (hunk 4 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 3 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 2 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 1 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: Makefile (hunk 8 above)
Comment (PatStiles) [reply Makefile]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 7 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 9 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
+# Setting Aggregator Address
+This doc contains a guide on how to call `setAggregator(address)` to set the `alignedAggregator` value of the AlignedServiceManager.sol contract. 
+
+### NOTE:
+- This guide assumes the Aligned layer contracts have been sucessfully deployed and the deployment outputs are within `./contracts/script/output/<DEPLOYMENT_FOLDER>`
+
+## 1.) Locate the deployed Aligned Aggregator Address
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 1. Locate the deployed Aligned Aggregator Address
```

File: docs/0_internal/setting_aggregator_address.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+The address of Aligned Aggregator can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within:
+```
+    "permissions": {
+        ...
+        "alignedLayerAggregator": "<AGGREGATOR_ADDRESS",
+        ...
+    }
+```
+
+## 2.) Locate the Aligned Service Manager Address
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 2. Locate the Aligned Service Manager Address
```

File: docs/0_internal/setting_aggregator_address.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+The address of Aligned Service Manager can be found in `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` within:
+```
+    "addresses": {
+        ...
+        "alignedLayerServiceManager": "<ALIGNED_SERVICE_MANAGER_ADDRESS>",
+        ...
+    }
+```
+
+## 3.) Set Environment Variables
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 3. Set Environment Variables
```

File: docs/0_internal/setting_aggregator_address.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+To run the make targets specified in this guide, you must first set the following env vars within `./contracts/scripts/.env.<NETWORK>`:
+```
+RPC_URL=<rpc_url>
+PRIVATE_KEY=<aligned_service_manager_pauser_private_key>
+ALIGNED_SERVICE_MANAGER_ADDRESS=<aligned_service_manager_address>
+```
+
+## 4.) Check the current value of `alignedAggregator` within AlignedServiceManager.sol
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 4. Check the current value of `alignedAggregator` within AlignedServiceManager.sol
```

File: docs/0_internal/setting_aggregator_address.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+cast call --rpc-url <RPC_URL> $ALIGNED_SERVICE_MANAGER_ADDRESS "alignedAggregator()(address)"
+```
+
+You should see that the returned address matches the returned address matches the address from `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` 
+
+## 5.) Change the value of `alignedAggregator` within AlignedServiceManager.sol
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 5. Change the value of `alignedAggregator` within AlignedServiceManager.sol
```

File: docs/0_internal/setting_aggregator_address.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+Set the environment variable `AGGREGATOR_ADDRESS` to the new address of the aggregator.
+```
+export AGGREGATOR_ADDRESS=<aggregator_address>
+make set_aggregator_address
+```
+
+## 6.) Verify the Aligned Aggreagtor Address has changed
```
Comment (avilagaston9) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
## 6. Verify the Aligned Aggreagtor Address has changed
```

**Review by avilagaston9 — APPROVED** [review]
Left some nit comments. Otherwise, LGTM!

File: docs/0_internal/setting_aggregator_address.md (hunk 10 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 11 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 12 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 13 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 14 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md (hunk 15 above)
Comment (PatStiles) [reply docs/0_internal/setting_aggregator_address.md]: addressed!

File: docs/0_internal/setting_aggregator_address.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+ALIGNED_SERVICE_MANAGER_ADDRESS=<aligned_service_manager_address>
+```
+
+## 4. Check the current value of `alignedAggregator` within AlignedServiceManager.sol
+
+```
+cast call --rpc-url <RPC_URL> $ALIGNED_SERVICE_MANAGER_ADDRESS "alignedAggregator()(address)"
+```
+
+You should see that the returned address matches the returned address matches the address from `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json`
```
Comment (JulianVentura) [inline docs/0_internal/setting_aggregator_address.md]:
```suggestion
You should see that the returned address matches the address from `./contracts/script/output/<DEPLOYMENT_FOLDER>/aligned_deployment_output.json` 
```

File: docs/0_internal/setting_aggregator_address_multisig.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
+# Propose the Transaction for Setting Aggregator Address using Multisig
+
+If you want to unpause the contracts, you can propose the unpause transaction using the multisig wallet.
```
Comment (uri-99) [inline docs/0_internal/setting_aggregator_address_multisig.md]:
```suggestion
If you want to set the Aggregator Address, you can propose the set Aggregator Address transaction using the multisig wallet.
```

File: docs/0_internal/setting_aggregator_address_multisig.md [hunk 18]
```diff
@@ -0,0 +1,55 @@
+# Propose the Transaction for Setting Aggregator Address using Multisig
+
+If you want to set the Aggregator Address, you can propose the set Aggregator Address transaction using the multisig wallet.
+
+## Prerequisites
+
+- You need to have deployed the contracts following the [Deploy Contracts Guide](./2_deploy_contracts.md).
+
+## Propose transaction for UnPause AlignedLayerServiceManager
```
Comment (uri-99) [inline docs/0_internal/setting_aggregator_address_multisig.md]:
```suggestion
## Propose transaction for Set Aggregator Address
```

File: docs/0_internal/setting_aggregator_address_multisig.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+
+   ![Choose pause](./images/set_aggregator_address_5.png)
+
+7. Click on `+ Add new transaction`
+
+   You should see the new transaction to be executed
+
+8. Click on `Create batch` to create the transaction.
+
+   ![Choose pause](./images/set_aggregator_address_6.png)
```
Comment (uri-99) [inline docs/0_internal/setting_aggregator_address_multisig.md]:
```suggestion
   ![Set Aggregator](./images/set_aggregator_address_6.png)
```

File: docs/0_internal/setting_aggregator_address_multisig.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+
+   ![Choose pause](./images/set_aggregator_address_5.png)
+
+7. Click on `+ Add new transaction`
+
+   You should see the new transaction to be executed
+
+8. Click on `Create batch` to create the transaction.
+
+   ![Set Aggregator](./images/set_aggregator_address_6.png)
```
Comment (uri-99) [inline docs/0_internal/setting_aggregator_address_multisig.md]:
```suggestion
   ![Choose set_aggregator](./images/set_aggregator_address_5.png)

7. Click on `+ Add new transaction`

   You should see the new transaction to be executed

8. Click on `Create batch` to create the transaction.

   ![create batch](./images/set_aggregator_address_6.png)
```

File: Makefile [hunk 21]
```diff
@@ -602,6 +602,16 @@ upgrade_add_aggregator: ## Add Aggregator to Aligned Contracts
...
```
Comment (uri-99) [inline Makefile]:
```suggestion
set_aggregator_address:
```

File: Makefile [hunk 22]
```diff
@@ -602,6 +602,16 @@ upgrade_add_aggregator: ## Add Aggregator to Aligned Contracts
 	@echo "Adding Aggregator to Aligned Contracts..."
 	@. contracts/scripts/.env && . contracts/scripts/upgrade_add_aggregator_to_service_manager.sh
 
+set_aggregator_address:
+	@echo "Setting Aggregator Address in Aligned Service Manager Contract on $(NETWORK) network..."
+	@echo "Aggregator address: $(AGGREGATOR_ADDRESS)"
+	@. contracts/scripts/.env.$(NETWORK) && . contracts/scripts/set_aggregator_address.sh $(AGGREGATOR_ADDRESS)
+
+set_aggregator_address_devnet: ## Add Aggregator to Aligned Contracts
```
Comment (uri-99) [inline Makefile]:
```suggestion
set_aggregator_address_devnet:
```

File: docs/0_internal/7_b_1_setting_aggregator_address_multisig.md
Comment (MarcosNicolau) [inline docs/0_internal/7_b_1_setting_aggregator_address_multisig.md:1]: I would add the verification with `cast` here at the end: 

```
cast call --rpc-url <RPC_URL> $ALIGNED_SERVICE_MANAGER_ADDRESS "alignedAggregator()(address)" 
```

File: docs/0_internal/7_asetting_aggregator_address.md
Comment (MarcosNicolau) [inline docs/0_internal/7_asetting_aggregator_address.md:1]: The file name is wrong

**Review by MarcosNicolau — APPROVED** [review]
Works for me!

**Review by JulianVentura — APPROVED** [review]
Worked as expected. Great work!

**Review by JuArce — APPROVED** [review]

**Review by uri-99 — APPROVED** [review]

**Pull request merged**